src/HotChocolate/Language/src/Language.SyntaxTree/Utilities/SyntaxPrinter.cs

```diff
@@ -65,9 +65,9 @@ public static string Print(this ISyntaxNode node, bool indented = true)
         CancellationToken cancellationToken = default)
     {
 #if NETSTANDARD2_0
-        using var streamWriter = new StreamWriter(stream, Encoding.UTF8);
+        using var streamWriter = new StreamWriter(stream, new UTF8Encoding(false));
```

Comment:
The documentation recommends that `throwOnInvalidBytes` is set to `true`:

https://learn.microsoft.com/en-us/dotnet/api/system.text.utf8encoding?view=net-8.0

> To enable error detection and to make the class instance more secure, you should call the [UTF8Encoding(Boolean, Boolean)](https://learn.microsoft.com/en-us/dotnet/api/system.text.utf8encoding.-ctor?view=net-8.0#system-text-utf8encoding-ctor(system-boolean-system-boolean)) constructor and set the throwOnInvalidBytes parameter to true. With error detection enabled, a method that detects an invalid sequence of characters or bytes throws an [ArgumentException](https://learn.microsoft.com/en-us/dotnet/api/system.argumentexception?view=net-8.0) exception. Without error detection, no exception is thrown, and the invalid sequence is generally ignored.

https://learn.microsoft.com/en-us/dotnet/api/system.text.utf8encoding.-ctor?view=net-8.0

> For security reasons, you should enable error detection by calling a constructor that includes a throwOnInvalidBytes parameter and setting its value to true.

I've done that for #6751.